### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by hexjelly** [hexjelly/elma-rust#2]
Got some quick code reviews after having finished the level parsing in [this weekly Rust thread](https://www.reddit.com/r/rust/comments/4e8ryn/hey_new_rust_users_got_an_easy_question_ask_here/d20hu4y) on Reddit. Try to implement some of this and use slices, ditch the Cursor and the ghetto function with a Reader trait (?) that I made.

**Comment by hexjelly** [comment]
Started writing some of this on [the refactor branch](https://github.com/hexjelly/elma-rust/tree/refactor).

Was mostly smooth sailing, until I got to the crypt_top10 function, where after asking for help in #rust-beginners I realized I'd have to use as_mut_slice() throughout the whole parse method. Which is now giving me trouble in the foor loop for the pictures, as I can't use split_at_mut() inside it. I think? Now in hindsight, it might be worthwhile to make the crypt_top10 function not mutate the actual raw binary data like I'm doing now, and instead leave it in place and do something like to_vec() on it and just return a Vec, or a slice from that... Seeing as the raw field will be overwritten after making any edits to a level and saving it anyway, there is no value to having the decrypted data there.Ugh who knows.
